src/controllers/game.controller.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  Controller,
+  Post,
+  Body,
+  BadRequestException,
+  Logger,
+} from '@nestjs/common';
+import { ApiTags, ApiResponse } from '@nestjs/swagger';
+import { GamesService } from 'src/Hangman/Application/Services/games.service';
+
+@Controller('games')
+@ApiTags('Games')
+export class GamesController {
+  private readonly logger = new Logger(GamesController.name);
+
+  constructor(private readonly gameService: GamesService) {}
+
+  @ApiResponse({ status: 201, description: 'Game created' })
+  @Post()
+  async startNewGame(
+    @Body()
+    {
+      gameId,
+      playerId,
+      wordToGuess,
+      maxGuesses,
+    }: {
+      gameId: string;
+      playerId: string;
+      wordToGuess: string;
+      maxGuesses: string;
+    },
+  ) {
+    try {
+      return await this.gameService.startNewGame(
+        gameId,
+        playerId,
+        wordToGuess,
+        parseInt(maxGuesses),
+      );
+    } catch (err) {
+      this.logger.error('startNewGame', err);
+
+      throw new BadRequestException("Can't start a new game");
```

Comment:
now realise: the error thrown in Aggregate is caught here, and I can log that for my own logs, but no reason to pass that to the user.

after that logging, I throw a BadRequestException, which is unhandled, so it goes through my filter where I can customize the response plus log again if I want.

Reply:
do I want to pass the reason for failing (the message in the Exception that was thrown in the Aggregate) to the user or not for security reasons?

Reply:
Rule of thumb is don't log sensitive data. In case of a hangman game there is no real sensitive data. We you could tell the end user is: "Could not <commandName-here>, domain requirement not met. This issue is logged". Such message gives a good indication what is going on but does not tell which requirement is not meant.

But remember every use case is different.

Reply:
ok. in dit geval geven we `err` die je in de `catch` krijgt dus niet door aan de BadRequestException richting de user.